README.adoc

```diff
@@ -91,6 +92,57 @@ When a rule matches, the Authorizer will return that rule’s evaluation as a de
 When no rules match, then by default it is denied.
 So, if ACL is not defined for a resource type then it automatically gets denied.
 
+== ManagedKafkaCreateTopicPolicy
+
+The objective of this policy is to:
+
+* validate the topic replication factor (default allows 3 only)
+* validate number of partitions in the cluster
+* validate the create topic configs
+
+To config the create topic policy, you should add config in Strimzi's Kafka CR.
+A sample CR fragment looks like:
+
+[source,yaml]
+----
+kafka:
+  spec:
+    config:
+      create.topic.policy.class.name=io.bf2.kafka.topic.ManagedKafkaCreateTopicPolicy
+      alter.config.policy.class.name=io.bf2.kafka.config.ManagedKafkaAlterConfigPolicy
+      # partition limit setting
+      kas.policy.create-topic.max.partitions=1000
+      kas.policy.create-topic.partition-counter.timeout-seconds=10
+      kas.policy.create-topic.partition-counter.private-topic-prefix="__redhat"
+      kas.policy.create-topic.partition-counter.schedule-interval-seconds=15
+      kas.policy.create-topic.partition-limit-enforced=true
+      # topic config setting
+      kas.policy.topic-config.enforced=compression.type:producer,segment.jitter.ms:0
+      kas.policy.topic-config.mutable=cleanup.policy,delete.retention.ms,retention.bytes,retention.ms
+      kas.policy.topic-config.range=segment.bytes:52428800:,segment.ms:600000:
+----
+
+Here are the configuration description:
+
+* `kas.policy.create-topic.max.partitions`: Used to specify the upper limit of partitions that should be allowed in the cluster. If this property is not specified, a default of -1 (no limit) is set
+* `kas.policy.create-topic.partition-counter.timeout-seconds`: Used to specify the number of seconds to use as a timeout duration when listing and describing topics method. Default is 10.
+* `kas.policy.create-topic.partition-counter.private-topic-prefix`: Used to specify the topic prefix to match for private/internal topics. Default is no prefix.
+* `kas.policy.create-topic.partition-counter.schedule-interval-seconds`: Used to specify the interval (in seconds) at which to schedule partition counts. Default is 15.
+* `kas.policy.create-topic.partition-limit-enforced`: Used to allow disabling of partition limit enforcement. Default is false.
+* `kas.policy.topic-config.enforced`: Used to specify the configs with permitted value. It's a list in the form [keyA]:[valueA],[keyB]:[valueB]. Default value is: "flush.ms:9223372036854775807,index.interval.bytes:4096,compression.type:producer,flush.messages:9223372036854775807,segment.jitter.ms:0,min.cleanable.dirty.ratio:0.5,file.delete.delay.ms:60000,segment.index.bytes:10485760,preallocate:false,unclean.leader.election.enable:false"
```

Comment:
```suggestion
* `kas.policy.topic-config.enforced`: Specifies a mapping between a topic policy key and a value. The property is expressed as a comma separated list of `key:value` pairs where the key and value are separated by  a `:`. Hence the it takes the form `keyA:value1,keyB:value2`. Default value is: "flush.ms:9223372036854775807,index.interval.bytes:4096,compression.type:producer,flush.messages:9223372036854775807,segment.jitter.ms:0,min.cleanable.dirty.ratio:0.5,file.delete.delay.ms:60000,segment.index.bytes:10485760,preallocate:false,unclean.leader.election.enable:false"
```

I found the use of `[]` around the key and value in the format description confusing as its trying to describe the parsing rules including extra tokens felt misleading. I'm easily persuaded that everyone knows they imply optional and thus should be included...

Reply:
Good suggestion.